GiTracker/Services/Rest/IRestService.cs

```diff
@@ -6,7 +7,10 @@ namespace GiTracker.Services.Rest
 {
     public interface IRestService
     {
-        Task<object> GetAsync(string host, string url, Type responseType, CancellationToken cancellationToken);
-        Task<T> GetAsync<T>(string host, string url, CancellationToken cancellationToken);
+        Task<object> GetAsync(string host, string url, Dictionary<string, string> parameters,
+            Type responseType, CancellationToken cancellationToken);
+
+        Task<T> GetAsync<T>(string host, string url, Dictionary<string, string> parameters,
+            CancellationToken cancellationToken);
     }
 }
```

Comment:
What if we have Task<object> GetAsync(RequestSettings requestSettings, CancellationToken cancellationToken)?
Then IGitApiProvider instead of having Host and Url, and request params will only provide one RequestSettings object for each request.